dashboard/app/models/user.rb

```diff
@@ -1081,10 +1081,21 @@ def self.authenticate_with_section_and_secret_picture(section:, params:)
     )
   end
 
+  # There is a bug (fix: https://codedotorg.atlassian.net/browse/INF-571) where some users have
+  # duplicate user levels for the same level. To ensure that we return the relevant user level for
+  # each level and not one of the duplicates, the list is first sorted so that the
+  # most relevant user levels are at the end. The list is then indexed by level ID, which will
+  # pick up the last matching user level in the list.
+  def self.index_user_levels_by_level_id(user_levels)
+    # The correct user level is the one most recently updated or the first created
+    relevant_user_levels_last = user_levels.sort {|a, b| [a.updated_at, b.id] <=> [b.updated_at, a.id]}
+    relevant_user_levels_last.index_by(&:level_id)
```

Comment:
I could replace these two lines with a custom index_by where I do the comparison while iterating to build the resulting list, which would probably be more efficient, but this is more concise. Let me know if you have opinions.